File: Sentinela.TypeScript/html/componente/DivServidorTile.ts
/// <reference path="../../../Web.TypeScript/html/componente/ComponenteHtml.ts"/>
/// <reference path="../../../Web.TypeScript/html/Div.ts"/>
/// <reference path="../../../Web.TypeScript/Utils.ts"/>

module Sentinela
{
    // #region Importações

    import ComponenteHtml = Web.ComponenteHtml;
    import Div = Web.Div;
    import Utils = Web.Utils;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class DivServidorTile extends ComponenteHtml
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        private _actHome: ActHome;
        private _divStatus: Div;
        private _divTitulo: Div;
        private _objServidor: ServidorDominio;

        private get actHome(): ActHome
        {
            return this._actHome;
        }

        private set actHome(actHome: ActHome)
        {
            this._actHome = actHome;
        }

        private get divStatus(): Div
        {
            if (this._divStatus != null)
            {
                return this._divStatus;
            }

            this._divStatus = new Div(this.strId + "_divStatus");

            return this._divStatus;
        }

        private get divTitulo(): Div
        {
            if (this._divTitulo != null)
            {
                return this._divTitulo;
            }

            this._divTitulo = new Div(this.strId + "_divTitulo");

            return this._divTitulo;
        }

        private get objServidor(): ServidorDominio
        {
            return this._objServidor;
        }

        private set objServidor(objServidor: ServidorDominio)
        {
            this._objServidor = objServidor;
        }

        // #endregion Atributos

        // #region Construtor

        constructor(actHome: ActHome, objServidor: ServidorDominio)
        {
            super(null);

            this.actHome = actHome;
            this.objServidor = objServidor;
            this.strId = (DivServidorTile.name + "_" + this.intObjetoId);
        }

        // #endregion Construtor

        // #region Métodos

        protected montarLayout(): void
        {
            super.montarLayout();

            this.divTitulo.strConteudo = this.objServidor.strNome;
        }

        protected montarLayoutFixo(strLayoutFixo: string): string
        {
            strLayoutFixo = super.montarLayoutFixo(strLayoutFixo);

            strLayoutFixo = Utils.replaceAll(strLayoutFixo, "_div_id", this.strId);

            return strLayoutFixo;
        }

        // #endregion Métodos

        // #region Eventos
        // #endregion Eventos
    }
}

File: Principal/Html/Componente/DivServidorTile.cs
using NetZ.Web.Html;
using NetZ.Web.Html.Componente;
using NetZ.Web.Server.Arquivo.Css;

namespace Sentinela.Html.Componente
{
    internal class DivServidorTile : ComponenteHtml
    {
        #region Constantes

        #endregion Constantes

        #region Atributos

        private Div _divStatus;
        private Div _divTitulo;

        private Div divStatus
        {
            get
            {
                if (_divStatus != null)
                {
                    return _divStatus;
                }

                _divStatus = new Div();

                return _divStatus;
            }
        }

        private Div divTitulo
        {
            get
            {
                if (_divTitulo != null)
                {
                    return _divTitulo;
                }

                _divTitulo = new Div();

                return _divTitulo;
            }
        }

        #endregion Atributos

        #region Construtores

        #endregion Construtores

        #region Métodos

        protected override void inicializar()
        {
            base.inicializar();

            this.strId = "_div_id";
        }

        protected override void montarLayout()
        {
            base.montarLayout();

            this.divTitulo.setPai(this);

            this.divStatus.setPai(this);
        }

        protected override void setCss(CssArquivoBase css)
        {
            base.setCss(css);

            this.addCss(css.setBorderBottom(1, "solid", "white"));
            this.addCss(css.setPaddingBottom(10));
            this.addCss(css.setPaddingTop(10));
            this.addCss(css.setPosition("relative"));

            this.divStatus.addCss(css.setBackgroundColor("#4CAF50"));
            this.divStatus.addCss(css.setBorder(3, "solid", "#81C784"));
            this.divStatus.addCss(css.setBorderRadius(50, "%"));
            this.divStatus.addCss(css.setHeight(20));
            this.divStatus.addCss(css.setPosition("absolute"));
            this.divStatus.addCss(css.setRight(0));
            this.divStatus.addCss(css.setTop(5));
            this.divStatus.addCss(css.setWidth(20));

            this.divTitulo.addCss(css.setFontSize(20));
        }

        protected override void setStrId(string strId)
        {
            base.setStrId(strId);

            this.divStatus.strId = (strId + "_divStatus");
            this.divTitulo.strId = (strId + "_divTitulo");
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}